firefox_puppeteer/ui/update_wizard/dialog.py

```diff
@@ -0,0 +1,126 @@
+
+        :returns: A dictionary with information about the active patch
+        """
+        patch = self._software_update.patch_info
+        patch['download_duration'] = self._download_duration
+        return patch
+
+    def download(self, wait_for_finish=True, timeout=TIMEOUT_UPDATE_DOWNLOAD):
+        """ Download the update.
+
+        :param wait_for_finish: Optional, if True the function has to wait
+         for the download to be finished, default to `True`
+        :param timeout: Optional, How long to wait for the download to finish,
+         default to 360 seconds
+        """
+        self._prefs.set_pref(self.PREF_APP_UPDATE_ALTWINDOWTYPE, self.window_type)
+
+        try:
+            channel = self._software_update.update_channel
+            assert channel.default_channel == channel.channel, \
+                'The update channel has been set correctly. ' \
+                'default_channel: is {}, while channel is: {}'.format(
+                    self._software_update.update_channel.default_channel,
+                    self._software_update.update_channel.channel)
+
+            # If updates have already been found, proceed to download
+            if self.wizard.selected_panel in (self.wizard.updates_found_basic,
+                                              self.wizard.updates_found_billboard,
+                                              self.wizard.error_patching,
+                                              ):
+                self.select_next_page()
+
+            # If incompatible add-on are installed, skip over the wizard page
+            if self.wizard.selected_panel == self.wizard.incompatible_list:
+                self.select_next_page()
+
+            # Updates were stored in the cache, so no download is necessary
+            if self.wizard.selected_panel in (self.wizard.finished,
+                                              self.wizard.finished_background,
+                                              ):
+                pass
+
+            # Download the update
+            elif self.wizard.selected_panel == self.wizard.downloading:
+                if wait_for_finish:
+                    start_time = datetime.now()
+                    self.wait_for_download_finished(timeout)
+                    self._download_duration = datetime.now() - start_time
+
+                    Wait(self.marionette).until(
+                        lambda _: self.wizard.selected_panel in (self.wizard.finished,
+                                                                 self.wizard.finished_background,
+                                                                 ),
+                        message='Final wizard page has been selected.')
+
+            else:
+                raise Exception('Invalid wizard page for downloading an update: {}'.format(
+                                self.wizard.selected_panel))
+
+        finally:
+            self._prefs.restore_pref(self.PREF_APP_UPDATE_ALTWINDOWTYPE)
```

Comment:
We cannot leave this pref enabled all the time. It could cause problems with other test steps if this method is used outside of the update tests. So I strongly propose to leave this in here.